modules/tensor_mechanics/test/tests/generalized_plane_strain/generalized_plane_strain_ref_resid.i

```diff
@@ -0,0 +1,154 @@
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+[]
+
+[Mesh]
+  file = square.e
+[]
+
+[Problem]
+  type = ReferenceResidualProblem
+  reference_vector = 'ref'
+  extra_tag_vectors = 'ref'
+  group_variables = 'disp_x disp_y'
+[]
+
+[Variables]
+  [./scalar_strain_zz]
+    order = FIRST
+    family = SCALAR
+  [../]
+[]
+
+[AuxVariables]
+  [./temp]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./saved_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./saved_y]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[Postprocessors]
+  [./react_z]
+    type = MaterialTensorIntegral
+    rank_two_tensor = stress
+    index_i = 2
+    index_j = 2
+  [../]
+[]
+
+[Modules/TensorMechanics/Master]
+  [./all]
+    strain = SMALL
+    add_variables = true
+    displacements = 'disp_x disp_y'
+    generate_output = 'stress_xx stress_xy stress_yy stress_zz strain_xx strain_xy strain_yy strain_zz'
+    planar_formulation = GENERALIZED_PLANE_STRAIN
+    eigenstrain_names = eigenstrain
+    scalar_out_of_plane_strain = scalar_strain_zz
+    temperature = temp
+    extra_vector_tags = 'ref'
+  [../]
+[]
+
+[AuxKernels]
+  [./tempfuncaux]
+    type = FunctionAux
+    variable = temp
+    function = tempfunc
+    use_displaced_mesh = false
+  [../]
+  [./saved_x]
```

Comment:
It runs fine if I don't include these AuxKernels, and the only reason I'm doing so is so I can diff against the original .e file.  It segfaults when I have it set up like this now.  I'm looking into what's going on, but I might need your help @lindsayad.

Reply:
K I'll look at it in a bit if you don't figure it out

Reply:
I figured it out.  Looks like TagVectorAux only works if you do `execute_on = timestep_end`. The vector `_v` in that class has size zero the first time it gets called. It must be a known issue because all of the tests that use TagVectorAux have `execute_on=timestep_end`. If you don't do that, it segfaults in opt, or hits an assertion about accessing an element of the array out of bounds in dbg.

This fix gets the test passing, but that should be addressed.